### PR TITLE
Call `BridgeUIBatchInstanceCallback::onBatchComplete` after loading bundles synchronously

### LINKED ISSUE #10255: Bundles generated offline do not start running the RN app when executed (white screen on version 0.69+)
### Problem Description

I have build an app `base64string-package-maker` to be able to decrypt packages so that I could use the package on other apps.

The problem is that When I build `publish/create app Package` with `release` as a target I only get blank app.

That mean when I run `Install.ps1` the app gets installed but there is no js included what so ever.

I think that the app boundle is not included somehow or maybe something else.

I do not get Any exception or error, So I have no Idea what is happning.

### Steps To Reproduce

1- create an app and then publish it with release 
2- You will get a clean app without any js

OR

1- fork my app [base64string-package-maker](https://github.com/AlenToma/base64string-package-maker)
2- publish it and see that there is no js included

### Expected Results

The app should show the js components and run as normal.

### CLI version

8.0.3

### Environment

```markdown
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-3820 CPU @ 3.60GHz
    Memory: 5.70 GB / 11.92 GB
  Binaries:
    Node: 14.18.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.15 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: AI-212.5712.43.2112.8609683
    Visual Studio: 16.11.32602.291 (Visual Studio Enterprise 2019)
  Languages:
    Java: 18.0.1.1 - C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.1 => 0.69.1
    react-native-windows: 0.69.1 => 0.69.1
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.15063

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Release

### Snack, code example, screenshot, or link to a repository

[base64string-package-maker](https://github.com/AlenToma/base64string-package-maker)

### PR DESCRIPTION
**Pull request by jonthysell**
`JsToNativeBridge::callNativeModules` only calls the stored `BridgeUIBatchInstanceCallback::onBatchComplete` if it's at the end of a batch, and either a native/turbo module method had been called previously, or if there are new incoming method calls. This call to `onBatchComplete` is what causes the batched UI queue to be processed, rendering changes in the native UI.

However, some (simple) JS apps, such as the default template for a new RN app, don't make any native function calls during app startup. When loading bundles asynchronously from Metro, the `onBatchComplete` call to process the UI queue gets posted to the native queue, so it gets run appropriately.

However, when loading bundles synchronously (from a pre-built bundle file), the first `onBatchComplete` will only be called if a native/turbo module was called during app startup. Without it, the app is *technically running*, but batched UI queue just sits there with the tasks to build the initial UI unprocessed.

The result is a "blank app", i.e. the `ReactRootView` has no children. "Technically" the app has loaded and is running (if you click we'll dispatch events to the JS side).

This PR fixes the issue by explicitly calling the `onBatchComplete` after the app has loaded from a synchronous bundle load. If the batch has already been processed, then it's essentially a slightly wasteful no-op.

I have tested that this resolves the issue for the "simple" apps, and seems to have no adverse effect on apps that have already caused `onBatchComplete` to be called normally.

Closes #10255

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10296)

### CODE REVIEW

**Comment by jonthysell** [comment]
FYI, a more targeted fix would be to add a call to `m_delegate->recordTurboModuleAsyncMethodCall()` [here](https://github.com/facebook/react-native/blob/1237952d070bfe28ff2c94cd9bf8ea19e6cdd395/ReactCommon/cxxreact/NativeToJsBridge.cpp#L165) after `NativeToJsBridge::loadBundleSync` calls `m_executor->loadBundle()`.

This would simulate that a native/turbo module had been called by the bundle, and allow the `onBatchComplete` call to happen "naturally".

File: vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp [hunk 1]
```diff
@@ -611,8 +611,16 @@ void ReactInstanceWin::LoadJSBundles() noexcept {
...
               instanceWrapper->loadBundleSync(Mso::Copy(strongThis->JavaScriptBundleFile()));
+
               if (strongThis->State() != ReactInstanceState::HasError) {
                 strongThis->OnReactInstanceLoaded(Mso::ErrorCode{});
+
+                // When loading a bundle synchronously, if the bundle code never calls any native/turbo modules,
+                // then onBatchComplete() will never be called to process the first batch of UI tasks, meaning
+                // the app will never start rendering UI. So we manually call the callback here, just in case.
+                // See https://github.com/microsoft/react-native-windows/issues/10255
+                auto callback = std::make_unique<BridgeUIBatchInstanceCallback>(weakThis);
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp:622]: @vmoroz : When you get back from vacation: Will this need to be ported to the office Host as well?

**Comment by acoates-ms** [comment]
I have another fix for this (#10298), which will make the current batches complete correctly, instead of adding another batch.

**Comment by jonthysell** [comment]
Closing this in preference for #10298.

**Pull request closed without merging**